packages/amplify-provider-awscloudformation/resources/update-idp-roles-cfn.json

```diff
@@ -9,68 +9,85 @@
+              "                'Effect': 'Allow',",
+              "                'Principal': {'Federated': 'cognito-identity.amazonaws.com'},",
+              "                'Action': 'sts:AssumeRoleWithWebIdentity',",
+              "                'Condition': {",
+              "                    'StringEquals': {'cognito-identity.amazonaws.com:aud': idpId},",
+              "                    'ForAnyValue:StringLike': {'cognito-identity.amazonaws.com:amr': 'authenticated'}",
+              "                }",
+              "            }]",
+              "        };",
+              "        let unauthParamsJson = {",
+              "            'Version': '2012-10-17',",
+              "            'Statement': [{",
+              "                'Effect': 'Allow',",
+              "                'Principal': {'Federated': 'cognito-identity.amazonaws.com'},",
+              "                'Action': 'sts:AssumeRoleWithWebIdentity',",
+              "                'Condition': {",
+              "                    'StringEquals': {'cognito-identity.amazonaws.com:aud': idpId},",
+              "                    'ForAnyValue:StringLike': {'cognito-identity.amazonaws.com:amr': 'unauthenticated'}",
+              "                }",
+              "            }]",
+              "        };",
+              "        if (event.RequestType === 'Delete') {",
+              "            try {",
+              "                delete authParamsJson.Statement[0].Condition;",
+              "                delete unauthParamsJson.Statement[0].Condition;",
+              "                let authParams = {PolicyDocument: JSON.stringify(authParamsJson), RoleName: authRoleName};",
+              "                let unauthParams = {PolicyDocument: JSON.stringify(unauthParamsJson), RoleName: unauthRoleName};",
+              "                const iam = new IAMClient({region: event.ResourceProperties.region});",
+              "                let res = await Promise.all([",
+              "                    iam.send(new GetRoleCommand({RoleName: authParams.RoleName})),",
+              "                    iam.send(new GetRoleCommand({RoleName: unauthParams.RoleName}))",
+              "                ]);",
+              "                console.log('in res', res);",
+              "                res = await Promise.all([",
+              "                    iam.send(new UpdateAssumeRolePolicyCommand(authParams)),",
+              "                    iam.send(new UpdateAssumeRolePolicyCommand(unauthParams))",
+              "                ]);",
+              "                console.log('delete response data' + JSON.stringify(res));",
+              "                response.send(event, context, response.SUCCESS, {});",
+              "            } catch (err) {",
+              "                console.log(err.stack);",
+              "                response.send(event, context, response.SUCCESS, {Error: err});",
```

Comment:
this (i.e. `response.SUCCESS`) matches original behavior. 
This PR avoids major changes in logic flow and focuses on SDK upgrade and promise->async/await conversion.